modules/brushingandlinking/src/processors/brushingandlinkingprocessor.cpp

```diff
@@ -43,21 +49,59 @@ const ProcessorInfo BrushingAndLinkingProcessor::getProcessorInfo() const { retu
     : Processor()
+    , inport_("inport")
     , outport_("outport")
     , clearSelection_("clearSelection", "Clear Selection",
                       [&]() { outport_.getManager().clearSelected(); })
     , clearHighlight_("clearHighlight", "Clear Highlighting",
                       [&]() { outport_.getManager().clearHighlighted(); })
     , clearCols_("clearCols", "Clear Columns",
                  [&]() { outport_.getManager().clearSelected(BrushingTarget::Column); })
-    , clearAll_("clearAll", "Clear Selections and Highlights", [&]() {
-        outport_.getManager().clearSelected();
-        outport_.getManager().clearHighlighted();
-        outport_.getManager().clearSelected(BrushingTarget::Column);
-    }) {
+    , clearAll_("clearAll", "Clear Selections and Highlights",
+                [&]() {
+                    outport_.getManager().clearSelected();
+                    outport_.getManager().clearHighlighted();
+                    outport_.getManager().clearSelected(BrushingTarget::Column);
+                })
+    , logging_("logging", "Logging", false, InvalidationLevel::Valid)
+    , logFilterActions_("filterActions", "Filter Actions", true, InvalidationLevel::Valid)
+    , logSelectActions_("selectActions", "Select Actions", true, InvalidationLevel::Valid)
+    , logHighlightActions_("highlightActions", "Highlight Actions", true, InvalidationLevel::Valid)
+    , maxVisibleIndices_("maxVisibleIndices", "Max Visible Indices", 50, 1, 1000, 1,
+                         InvalidationLevel::Valid, PropertySemantics::Text) {
+
+    inport_.setOptional(true);
+    addPorts(inport_, outport_);
+
+    addProperties(clearSelection_, clearHighlight_, clearCols_, clearAll_, logging_);
+    logging_.addProperties(logFilterActions_, logSelectActions_, logHighlightActions_,
+                           maxVisibleIndices_);
+
+    outport_.getManager().setParent(&inport_.getManager());
+    outport_.getManager().onBrush([this](BrushingAction action, BrushingTarget target,
+                                         const BitSet& indices, std::string_view source) {
+        if (!logging_.isChecked()) return;
+        if ((action == BrushingAction::Filter) && !logFilterActions_) return;
+        if ((action == BrushingAction::Select) && !logSelectActions_) return;
+        if ((action == BrushingAction::Highlight) && !logHighlightActions_) return;
+
+        std::string str;
+        auto it = indices.begin();
+        int i = maxVisibleIndices_;
+        while (it != indices.end() && i > 0) {
+            str += std::to_string(*it++) + ',';
+            --i;
+        }
+        if (it != indices.end()) {
+            str += "...";
+        } else if (!str.empty()) {
+            str.pop_back();
+        }
 
-    addPort(outport_);
-    addProperties(clearSelection_, clearHighlight_, clearCols_, clearAll_);
+        LogProcessorInfo(fmt::format(
+            "{:<20} action: {}, target: {}, source: {}, indices: [{}] ({})", getDisplayName(),
```

Comment:
Do you need to put the display name here, don't we add that to the log anyway? 
Might also try and add some width to the other ones to align them?

Reply:
This was based on the event logger and old brushing events.

`LogProcessorInfo` will only print the identifier of the processor. Is this more useful than the display name?
https://github.com/inviwo/inviwo/blob/4567a9750a1efc3a0416ceb0bcb25ba0e96620a1/src/core/util/logcentral.cpp#L72-L76

Reply:
I opted for additional line break. In particular since target and source can be arbitrarily long. Could remove the identifier if you insist.